src/nativeMain/kotlin/api/EclairClient.kt

```diff
@@ -58,4 +63,22 @@ class EclairClient(private val apiHost: String, private val apiPassword: String)
             else -> ApiError(statusCode.value, "api error")
         }
     }
+
+    @OptIn(InternalAPI::class)
+    override suspend fun connect(uri: String): Either<ApiError, String> {
+        return try {
+            val response: HttpResponse = httpClient.submitForm(
+                url = "${apiHost}/connect",
+                formParameters = parameters {
+                    append("uri", uri)
+                }
+            )
```

Comment:
As per my understanding, my request is using multipart form data and the response is expected to be in JSON format. I'm using [submitForm](https://ktor.io/docs/request.html#form_parameters) with form parameters to send a multipart form data request. The ContentNegotiation plugin used above(outside this function, but in the same class)for specifying how to serialize/deserialize the requests/responses.The requests are using multipart form data, not JSON.

The submitForm call is manually handling the form data. What the ContentNegotiation plugin will do here is to automatically parse JSON responses, which is what I want. So, this also makes sense.

Ps: This is still not working as expected.
I'm still in process of applying what you told in this [comment](https://github.com/ACINQ/eclair-cli/pull/9#discussion_r1273186358).